### PR TITLE
feat: ENS lookup endpoint

### LINKED ISSUE #181: Initial ENS resolution endpoint

### LINKED ISSUE #182: Better ENS resolution error handling
Basic text errors were added in https://github.com/WalletConnect/rpc-proxy/pull/183, improve these.

E.g. NotFound when ENS is not found

### PR DESCRIPTION
**Pull request by chris13524**
# Description

This implements a basic ENS resolution endpoint. No caching has been implemented yet.

This PR also adds some configuration boilerplate for the Redis cache, but it is not in-use yet.

Resolves #181
Resolves #182

## How Has This Been Tested?

Locally with curl and against https://github.com/WalletConnect/web3modal/pull/1090

## Due Diligence

* [ ] Breaking change
* [ ] Requires a documentation update
* [ ] Requires a e2e/integration test update

### CODE REVIEW

File: src/handlers/identity.rs [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+) -> Result<Json<IdentityResponse>, RpcError> {
+    let provider = Provider::new(SelfProvider {
+        state,
+        connect_info,
+        query,
+        path,
+        headers,
+    });
+
+    let name = provider
```
Comment (arein) [inline src/handlers/identity.rs:51]: Can you later add something like Not Found?

File: src/handlers/identity.rs [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+    ) -> Result<R, Self::Error> {
+        debug!("Got SelfProvider request");
+
+        let id = SystemTime::now()
+            .duration_since(UNIX_EPOCH)
+            .expect("can get current time")
+            .as_millis()
+            .to_string();
+
+        let response = super::proxy::handler(
```
Comment (arein) [inline src/handlers/identity.rs:112]: Just confirming: this call will show up in the RPC HTTP metrics for error/provider attribution?

**Review by arein — APPROVED** [review]

File: src/handlers/identity.rs (hunk 2 above)
Comment (chris13524) [reply src/handlers/identity.rs:112]: Yup, all those metrics should be inside `super::proxy::handler()` already, so I'm just re-using that.

File: src/handlers/identity.rs (hunk 1 above)
Comment (chris13524) [reply src/handlers/identity.rs:51]: Yeah, I'll follow up before prod with improved error handling: https://github.com/WalletConnect/rpc-proxy/issues/182

Comment (chris13524) [reply src/handlers/identity.rs:55]: Just added better error handling

**Comment by arein** [comment]
Awesome!

**Pull request merged**